14/umbraco-cms/extending/health-check/guides/contentsecuritypolicy.md

```diff
@@ -0,0 +1,50 @@
+# Health check: Content Security Policy (CSP)
+
+_This check verifies if your site has a Content Security Policy (CSP) header to defend against Cross-Site Scripting (XSS) and data injection attacks._
+
+## How to fix this health check
+This health check can be fixed by adding a header before the response is started.
+
+Preferable you use a security library like [NWebSec](https://docs.nwebsec.com/).
+
+### Adding a Content Security Policy (CSP) using NWebSec
+
+If you take a NuGet dependency on [NWebsec.AspNetCore.Middleware/](https://www.nuget.org/packages/NWebsec.AspNetCore.Middleware/), you can use third extension methods on `IApplicationBuilder`.
+
+```csharp
+...
+WebApplication app = builder.Build();
+app.UseCsp(options => options
+    .ImageSources(s => s
+        .Self()
+        .CustomSources(
+            "our.umbraco.com data:",
+            "dashboard.umbraco.com"))
+    .DefaultSources(s => s
+        .Self()
+        .CustomSources(
+            "our.umbraco.com",
+            "marketplace.umbraco.com"))
+    .ScriptSources(s => s
+        .Self())
+    .StyleSources(s => s
+        .Self())
+    .FontSources(s => s
+        .Self())
+    .ConnectSources(s => s
+        .Self())
+    .FrameSources(s => s
+        .Self()));
+```
+
+### Adding a Content Security Policy (CSP) using manual middleware
+
+Avoid third-party library dependency by using custom middleware added to the request pipeline.
```

Comment:
Suggested a small rewrite here, to make it a little more clear what the code snippet below does.
Let me know if it does not make sense!

```suggestion
Avoid third-party library dependencies by using custom middleware added to the request pipeline as shown below.
```